llm/server.go

```diff
@@ -214,15 +214,23 @@ func NewLlamaServer(gpus discover.GpuInfoList, model string, ggml *GGML, adapter
 		params = append(params, "--threads", strconv.Itoa(defaultThreads))
 	}
 
-	flashAttnEnabled := envconfig.FlashAttention()
+	// Check flash attention support and determine KV cache type
+	flashAttnEnabled, reason := checkFlashAttentionSupport(ggml)
```

Comment:
In light of the other the comments in this review, I think we can land with something like this.

```
fa := envconfig.FlashAttention()
if fa && !gpus.FlashAttentionSupported() {
    slog.Warn("flash attention enabled but not supported by gpu")
    fa = false
}

if fa && !ggml.SupportsFlashAttention(ggml) {
    slog.Warn("flash attention enabled but not supported by model")
    fa = false
}

if fa {
    slog.Info("enabling flash attention")
    params = append(params, "--flash-attn")

    // flash attention also supports kv cache quantization
    // check if the requested kv cache type is supported by the
    // model and use it if so
    kvct := envconfig.KvCacheType()
    if ggml.SupportsKVCacheType(kvct) {
        params = append(params, "--kv-cache-type", kvct)
    } else {
        slog.Warn("kv cache type not supported by model", "type", kvct)
    }
}
```

Reply:
As per other comment, I've hopefully made the suggested changes.